### DISCUSSION TITLE
タグの先頭の # や最後の , を自動削除して欲しい

### DISCUSSION THREAD

**Discussion by shiracamus** [increments/qiita-discussions#67] [要望・アイデア・不具合]
Qiitaのタグは `#` なしが基本だと思っていますが、同じタグに`#`を付けて投稿されている記事が沢山あります。twitterなどで `#` 付に慣れている人も多いと思います。
Qiitaで`#`なしのタグをフォローしている場合、`#`付の新着記事がタイムラインに表示されません。
そこで、タグの先頭に指定された `#` や、最後に付けた `,` を自動削除して投稿されるようにして欲しいです。
既存記事は変更せずに読者が編集リクエストを出せばいいと思います。私の編集リクエストは大半がそれです。

* 自動削除する先頭文字 ： `#`, `＃`, `,`, `，`, `、` （時々全角文字で指定する人がいます）
* 自動削除する最終文字 ： `,`, `，`, `、`（カンマ区切りで指定する人がいるので）
* 空白に置換する中間文字 ： `,` （カンマ区切りで指定する人がいるので）

`#` で始めたいタグがあった場合、`##タグ` と指定して最初の1文字を削除して、`#タグ` で投稿されるようにしてはいかがでしょうか。
あるいは、「タグをこのように補正しますがよろしいですか？」と再編集を求めて、`#`付に直せるようにするとか。

あと、タグの用途も明確にして欲しいです。

* 同じ興味・関心を持つ人・記事を集めるための共通キーワード
* 検索キーワードではなく、カテゴリやジャンルを指定

とか。

**Comment by kskwtnk** [comment]
@shiracamus 

こちらも、ご意見ありがとうございます！

`#`や`,`がついている記事を確認したところかなりの数がありましたので、対応を検討いたします！
（`#`付きのタグを作りたい場合の挙動は少し考えることが多そうで、ご提案いただいた内容の通りにはならないかもしれません 🙏 ）

また、タグの用途についてはそこまで厳密に定義する予定は現在はありません
というのも各種SNSや他の情報共有サービスにおいてもある程度緩いルール決めになっており、その緩さが人々の良い繋がりを生み出すこともあるためです
例えばですが、もし実施するとしても`たくさんの人に読まれるためにはこういうタグの付け方がおすすめです`といったコミュニケーションに留まると思います 🙏

**Comment by shiracamus** [comment]
あと、バージョン番号だけのタグが指定されることもあって、そのタグでコミュニケーションすることはないと思うので、v+数字とドット  (正規表現: "[vVｖV]?[\d.]+" ) の指定も除外してはいかがでしょうか

**Reply by kskwtnk** [reply]
追加でありがとうございます！

こちらについては、現状除外をする予定はありません 🙏 

たしかに仰る通り数字だけのタグが機能するタイミングはほぼ100%存在しないと思うのですが、こういった内容を手動で除外をしていくと、どこかで「有用ともそうでないとも言えない」ラインがやってきてしまいます

有用なタグまで除外してはむしろ使い勝手が下がってしまうので、明確なデメリットを生むものではない以上触らないという判断をしております 🙏

**Reply by munierujp** [reply]
[v8](https://qiita.com/tags/v8)がマッチしてしまいますね。

**Reply by shiracamus** [reply]
8080 や 8086 などもあって自動化は難しそうですね。
地道に編集リクエストすることにします。

**Comment by kskwtnk** [comment]
@shiracamus 

こちらについてご意見ありがとうございました！
もともと運営が定期的に名寄せの処理を行っていたのですが、今回いただいたのをきっかけに`#`や`,`つきのタグを名寄せいたしました

このDiscussionとしてのご要望とは違った形式になってしまいましたが、対応しましたのでご報告です！

**Reply by shiracamus** [reply]
@xrxoxcxox 対応ありがとうございます。
全てのタグに対応したわけではないのですね。
記事のタグを見てみると、例えば ` Python, #DeepLearning, #Pytorch, #flow, #超解像` のように Python だけ `#` を取り除いているようですね。
今後もタグ修正の編集リクエストを続けることになりそうです。編集リクエストだけでユーザランキングTop10にいるのが申し訳ない気持ち・・・

**Reply by kskwtnk** [reply]
@shiracamus 

> 全てのタグに対応したわけではないのですね。

今回はまず記事数の多いものについて対応をしました！
そのため、1桁記事数くらいのタグは対応できておりません 🙏 

> 今後もタグ修正の編集リクエストを続けることになりそうです。

こちらこそ申し訳ない限りです…… 🙇 
我々も見つけ次第対応していきます！
